release_notes/ocp-4-11-release-notes.adoc

```diff
@@ -2457,6 +2438,38 @@ spec:
 
 * The pipeline metrics API does not support the required `pipelinerun/taskrun` histogram values from {rh-openstack} 1.6 and beyond. Consequently, the *Metrics* tab in the *Pipeline* -> *Details* page is removed instead of displaying incorrect values. There is currently no workaround for this issue. (link: https://bugzilla.redhat.com/show_bug.cgi?id=2074767[*BZ#2074767*])
 
+* In {product-title} {product-version}, pod security admission runs globally with privileged enforcement and restricted audit logging and API warnings. Versions of the `opm` binary released earlier than {product-title} {product-version}, do not work with the restricted profile setting. As a result, an error displays if you install a catalog source that was built with an `opm` binary released earlier than {product-title} {product-version}.
++
+.Example error
+[source,terminal]
+----
+Error: open ./db-xxxx: permission denied
+----
++
+Workaround: Cluster administrators can allow the creation of catalog sources built with earlier `opm` versions by setting the pod security admission labels in the namespace metadata files. Apply the `baseline` label to the `warn`, `audit`, and `enforce` modes and set the `scc.podSecurityLabelSync` label to `false`, as shown in the following example:
```

Comment:
@michaelryanpeter @anik120:

It is wrong:

a) All namespaces are sync by default with ``security.openshift.io/scc.podSecurityLabelSync` unless those that are considered Openshift system namespaces. Please, see the SCC doc: https://github.com/openshift/openshift-docs/pull/48409/files#diff-a9fd77a91dcedf5db05ac33bca1bcd89e2c8c2e4b444b6dbdf0f6c06cba1af53R19-R35
b) Therefore, we should NOT recommend users create CatalogSource or Operator in Openshift system namespaces
c) The namespace openshift-marketplace is ONLY used by OLM and OLM labels it.

So, we need to clarify it here:

a) If you want to create a new Catalogue Source it is not recommended use the Openshift system namespace
b) If you create a CatalogSource in a namespace prefixed with openshift and you face the error because you are using an image that was built with the old OPM tool then, ensure that you enable the PSA label synchronization. We should link  SCC doc here: https://github.com/openshift/openshift-docs/pull/48409/files#diff-6b079749a009ff430faa75c3cdece0921ba0e269d1dcd32d6ec0939c7ff99059R30-R44

@jianzhangbjz ^

Reply:
>  If you want to create a new Catalogue Source it is not recommended use the Openshift system namespace

If the customer create a CatalogSource in their namespace, they will encounter bug https://bugzilla.redhat.com/show_bug.cgi?id=2088541, the workaround is add labels for their namespace:
```yaml
apiVersion: v1
kind: Namespace
metadata:
...
  labels:
    security.openshift.io/scc.podSecurityLabelSync: "false"
    openshift.io/cluster-monitoring: "true"
    pod-security.kubernetes.io/audit: baseline <1>
    pod-security.kubernetes.io/warn: baseline <2>
    pod-security.kubernetes.io/enforce: baseline <3>
  name: "<namespace_name>"
```

Reply:
@jianzhangbjz the warning is NOT a bug. It was tracked as a bug with the expectation that all that is shipped/installed by default via OLM does not raise warnings. 

The workarround here is about the error `Error: open ./db-xxxx: permission denied` and not the warning.
Therefore, the workarround here is not right and we need to clarify that users should NOT create catalog sources in Openshuft system namespaces and if they create a new namespace prefixed with openshift they also should enable the PSA label syncer instead. 

c/c @michaelryanpeter @anik120

Reply:
> The workarround here is about the error Error: open ./db-xxxx: permission denied

Hi @camilamacedo86 , thanks! But, why do we need this workaround? Based on https://bugzilla.redhat.com/show_bug.cgi?id=2100323#c4, it had been fixed.

Reply:
This index image `quay.io/olmqe/etcd-index:v1` was created by using the old `opm`, but I don't encounter this `Error: open ./db-xxxx: permission denied` error when installing it on the latest OCP 4.11 cluster. As follows,
```yaml
MacBook-Pro:~ jianzhang$ oc get clusterversion
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.11.0-0.nightly-2022-08-04-081314   True        False         3h30m   Cluster version is 4.11.0-0.nightly-2022-08-04-081314
MacBook-Pro:~ jianzhang$ cat cs-test.yaml 
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: test-operators
  namespace: openshift-marketplace
spec:
  displayName: Jian Operators
  image: quay.io/olmqe/etcd-index:v1
  priority: -100
  publisher: Jian
  sourceType: grpc
  updateStrategy:
    registryPoll:
      interval: 10m0s

MacBook-Pro:~ jianzhang$ oc get catalogsource
NAME                  DISPLAY                TYPE   PUBLISHER      AGE
certified-operators   Certified Operators    grpc   Red Hat        3h46m
community-operators   Community Operators    grpc   Red Hat        3h46m
qe-app-registry       Production Operators   grpc   OpenShift QE   3h24m
redhat-marketplace    Red Hat Marketplace    grpc   Red Hat        3h46m
redhat-operators      Red Hat Operators      grpc   Red Hat        3h46m
test-operators        Jian Operators         grpc   Jian           2m23s

MacBook-Pro:~ jianzhang$ oc get pods
NAME                                                              READY   STATUS      RESTARTS        AGE
1263538589d348af98ee1c1a3af1e63c7d8f8f7b9bf1a043d121dac5c9c92fk   0/1     Completed   0               3h21m
3ef707a2bab80ae871eaa876bd55216b08d6a4980ebb5b8ed5cd9c123d7rpqf   0/1     Completed   0               3h21m
certified-operators-wnwzw                                         1/1     Running     0               3h20m
community-operators-5tr9l                                         1/1     Running     0               122m
marketplace-operator-5b84d4d799-4scrc                             1/1     Running     1 (3h39m ago)   3h50m
qe-app-registry-kl8gq                                             1/1     Running     0               3h22m
redhat-marketplace-4bgs9                                          1/1     Running     0               3h43m
redhat-operators-mc46k                                            1/1     Running     0               3h43m
test-operators-dl4xf                                              1/1     Running     0               12s
MacBook-Pro:~ jianzhang$ oc rsh test-operators-dl4xf
/ # ps -elf|cat
PID   USER     TIME  COMMAND
    1 root      0:00 /bin/opm registry serve --database /database/index.db
   78 root      0:00 /bin/sh
   84 root      0:00 ps -elf
   85 root      0:00 cat
/ # exit

```

Reply:
```yaml
MacBook-Pro:~ jianzhang$ oc get ns openshift-marketplace -o yaml
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    capability.openshift.io/name: marketplace
    include.release.openshift.io/ibm-cloud-managed: "true"
    include.release.openshift.io/self-managed-high-availability: "true"
    include.release.openshift.io/single-node-developer: "true"
    openshift.io/node-selector: ""
    openshift.io/sa.scc.mcs: s0:c17,c9
    openshift.io/sa.scc.supplemental-groups: 1000290000/10000
    openshift.io/sa.scc.uid-range: 1000290000/10000
    workload.openshift.io/allowed: management
  creationTimestamp: "2022-08-08T23:14:30Z"
  labels:
    kubernetes.io/metadata.name: openshift-marketplace
    olm.operatorgroup.uid/a93b3ad3-dc83-4801-908e-65b1de607ded: ""
    openshift.io/cluster-monitoring: "true"
    pod-security.kubernetes.io/audit: baseline
    pod-security.kubernetes.io/warn: baseline
  name: openshift-marketplace
  ownerReferences:
  - apiVersion: config.openshift.io/v1
    kind: ClusterVersion
    name: version
    uid: d9e8b196-691f-4e8e-92c8-c83338f46eec
  resourceVersion: "8735"
  uid: ecc126e9-1d9e-43bd-8c40-7cb81c566497
spec:
  finalizers:
  - kubernetes
status:
  phase: Active
```

Reply:
I also install it in a `restricted` project, but still not encounter this `Error: open ./db-xxxx: permission denied`. So, I'm confused why we treat this `Error: open ./db-xxxx: permission denied` as a known issue, it had been fixed.
```yaml
MacBook-Pro:~ jianzhang$ oc get catalogsource -n test
NAME             DISPLAY          TYPE   PUBLISHER   AGE
test-operators   Jian Operators   grpc   Jian        14s
MacBook-Pro:~ jianzhang$ oc get pods -n test
NAME                   READY   STATUS    RESTARTS   AGE
test-operators-k9wsc   1/1     Running   0          21s
MacBook-Pro:~ jianzhang$ 
MacBook-Pro:~ jianzhang$ oc get ns test -o yaml
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    openshift.io/description: ""
    openshift.io/display-name: ""
    openshift.io/sa.scc.mcs: s0:c27,c14
    openshift.io/sa.scc.supplemental-groups: 1000730000/10000
    openshift.io/sa.scc.uid-range: 1000730000/10000
  creationTimestamp: "2022-08-09T03:21:26Z"
  labels:
    kubernetes.io/metadata.name: test
    pod-security.kubernetes.io/audit: restricted
    pod-security.kubernetes.io/audit-version: v1.24
    pod-security.kubernetes.io/warn: restricted
    pod-security.kubernetes.io/warn-version: v1.24
  name: test
  resourceVersion: "104920"
  uid: b76b2aec-0d90-4b70-acaa-f762c78add67
spec:
  finalizers:
  - kubernetes
status:
  phase: Active
```

Reply:
You are right @jianzhangbjz.
This case can NOT be sorted out by the PSA label sync =  true. 

I just tested it out and the OCP matches the pod/image to restricted-v2 so OCP is unable to check that the image requires scale permissions and it fails in one of the impacted case scenarios.

You can ignore my comment and we need to say as workarround just enforce the namespace as privileged.

Reply:
@jiazha@redhat.com you need to apple the label to enforce restricted: 
Example: oc label --overwrite ns <my-catalog-ns> pod-security.kubernetes.io/enforce=restricted.

In your test, the NS is configured to WARNING and AUDIT based on the restricted policy but it is NOT with the enforce.  